### DISCUSSION TITLE
A suggestion to redesign the form view

### DISCUSSION THREAD

**Discussion by PatriaMinas** [pawelsalawa/sqlitestudio#4330] [Ideas]
I know there are probably many more important things to do. So forgive me for the exaggeration. I'm suggesting a redesign of the form view. This would allow productivity for small task automations, which do not depend on high-performance software development.

Redesigning the form view would look like this:

![crudiu](https://user-images.githubusercontent.com/95147314/145123900-f649e5d9-382a-460a-bf10-58982f98bad9.png)

**Comment by pawelsalawa** [comment]
This is not that simple. I mean I would also like to have form views like this. The problem is that some columns will sometimes have a very long values, sometimes multi-line values, sometimes text, but sometimes numbers ...and sometimes even a binary data.

Different kind of data has to be displayed in a different form widgets. Keep in mind that in SQLite3 every row, on a same column - the table can have different data type, so navigating between rows would have to dinamically change widgets used - according to what data is there. That would make the whole form "jump around" and flicker when navigating between rows.

Then SQLiteStudio also has to take into account, that data is not always available in the cell and it's hard to predict what data type user wants to use for particular cell yet.

User also can change representation (data type) of existing data, for example change text into number and that can be done using different widget tabs for different data types.

Current form layout is a compromise between usability and database capabilities.

That being said - new SQLite3 versions (yet to be released) will bring `STRICT` tables: https://sqlite.org/draft/stricttables.html
These tables will have columns with data types strictly constrained - like on most of other databases on the market. For such tables I want SQLiteStudio to optimize form views, but we will see when that happens.

**Reply by PatriaMinas** [reply]
Thank you very much for commenting on my suggestion. Really, I imagine it's a little tricky to organize this layout.

**Comment by PatriaMinas** [comment]
So does this news sound interesting for the future of form view? I just went to read it now.

Ref.: https://www.sqlite.org/releaselog/3_37_0.html
SQLite Release 3.37.0 On 2021-11-27
STRICT tables provide a prescriptive style of data type management, for developers who prefer that kind of thing.

**Reply by pawelsalawa** [reply]
That is precisely what I mentioned in the last paragraph of my answer above. Yes, with `STRICT` tables, the Form View can be arranged in a better way. Still, it requires work and there are other issues with higher priority at the moment.
